### PR TITLE
Eventhub Scaler: add new dapr checkpoint strategy to eventhub scaler

### PR DESCRIPTION
**Pull request by christle**
Hello,

Dapr uses a prefix for the Checkpoint blob "dapr-hubName-consumerGroup-". So here is another Checkpoint Strategy to support that case. 

I will open another PR to update the docs.

Relates to #3022

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
looking good, but I think this is duplicated https://github.com/kedacore/keda/pull/3139
Thanks for the contribution, but let's wait to check if the original PR is active or no

**Comment by christle** [comment]
Oh you are right. I didn’t see that parallel activity.

**Comment by tomkerkhove** [comment]
Can we close this PR in favor of https://github.com/kedacore/keda/pull/3139 @christle?

**Comment by christle** [comment]
Yes, currently, I test the e2e test on this base, but I could cherry-pick the e2e over so that we can close this PR

**Comment by tomkerkhove** [comment]
What works for you!

**Comment by christle** [comment]
@tomkerkhove I'm ready with the e2e test, but before the new test is green, the related image from the PR https://github.com/kedacore/test-tools/pull/99 should be merged. I've tested it locally with a custom image on my registry.

I've added the e2e test to my PR because I have no access rights to the other fork. I hope it is ok.
Tomorrow, I will create the docs PR.

**Comment by christle** [comment]
it seems that the required package "ghcr.io/kedacore/tests-azure-eventhub-dapr" isn't pushed

**Comment by JorTurFer** [comment]
I'll check, I guess that it's private as default :(

**Comment by JorTurFer** [comment]
It was private, I have already changed it+

**Comment by JorTurFer** [comment]
It seems like scaling in doesn't work, maybe the timeout is too small or the worker doesn't consume the messages correctly

**Comment by christle** [comment]
On my private setup, based on the "test-infrastructure" repo, I can run all event* tests without problems. The "scale in" needs around 10+ seconds. Currently, I don't see the difference. I see inside the pipeline run a lot of errors around missing eventhub infrastructure and some BlobNotFound exceptions regarding the storageAccount. The BlobNotFound could be the reason for the missing "scale in" because there is no checkpoint. But I don't get this on my private azure, and I have no further instruments for the underlying pipeline infrastructure or the container logs. Could it be that are some parallel e2e tests that cause the deletion of the eventhubs and Storage Blobs?

**Comment by JorTurFer** [comment]
> On my private setup, based on the "test-infrastructure" repo, I can run all event* tests without problems. The "scale in" needs around 10+ seconds. Currently, I don't see the difference. I see inside the pipeline run a lot of errors around missing eventhub infrastructure and some BlobNotFound exceptions regarding the storageAccount. The BlobNotFound could be the reason for the missing "scale in" because there is no checkpoint. But I don't get this on my private azure, and I have no further instruments for the underlying pipeline infrastructure or the container logs. Could it be that are some parallel e2e tests that cause the deletion of the eventhubs and Storage Blobs?

We use the random names to avoid conflicts between tests, so it shouldn't happen. Let me check the logs

**Comment by christle** [comment]
ok thank you, I will check another possible reason regarding different dapr versions

**Comment by JorTurFer** [comment]
I'm checking in depth the e2e test, and you aren't installing dapr, so basically the checkpoint probably isn't created, and the messages aren't consumed. The e2e cluster doesn't have any infra as default, all needed infrastructure should be installed/removed during the tests. I can't see either any dapr manifest in your e2e test.
I guess you have deployed those things manually into your testing cluster, could this be the issue?

**Comment by JorTurFer** [comment]
you have examples about how you can install dapr "globally" [here](https://github.com/kedacore/keda/blob/main/tests/utils/setup_test.go#L57-L131), where we install Workload Identity and AWS Role Assumptions, and you also have other samples about "test scoped" installation in other scalers e2e tests like Datadog or NewRelic, where we install their against in the scope of the test, and we remove them at the end

**Comment by christle** [comment]
i deploy dapr as single edge container within the deployment template

**Comment by christle** [comment]
I have created a cluster with the kedacore/test-infrastructure repo, and there it runs as expected, but I will do some further checks. I think as a second container within the deployment, it is very easy to set up and to clean up

**Comment by JorTurFer** [comment]
> i deploy dapr as single edge container within the deployment template

true, I didn't see it, sorry

**Comment by christle** [comment]
I've found the issue. I have an older dapr image. I put "ImagePullAllways" inside the template and now I could reproduce the pipeline issue. Ok, i will check that.

**Comment by JorTurFer** [comment]
let me trigger them again and check the container logs

**Comment by christle** [comment]
Ok, got it. And a little bit strange.

I have used the daprd:edge image. In my test setup, it works fine. I pulled the image view days ago. If I check storage checkpoint, i can confirm that the blob name is the new dapr behavior since 1.9. 

If I watch now under daprio/dapr and check the edge tag, i notice that they changed it yesterday. I tested this newer image on my local test and have the same issue as the pipeline because it uses the old dapr checkpoint naming schema.

If I use the last version tag 1.9.5-mariner image every works fine again. Seems like a mistake within the edge tag. I will push a change. could please start the pipeline again after that?

**Comment by JorTurFer** [comment]
wow, nice catch! 👏 
I have triggered it again

**Review by JorTurFer — APPROVED** [review]
LGTM! Thanks for the improvement 🙇

**Comment by JorTurFer** [comment]
Could you create another PR to docs repo adding this new checkpoint?

**Comment by christle** [comment]
Yes, I will do it

**Comment by christle** [comment]
btw, I like the hole test setup very much, it was very easy to setup a test environment with the test-tools and test-infrastructure repo.

**Comment by christle** [comment]
I've created a pull request for the docs repository https://github.com/kedacore/keda-docs/pull/996

**Comment by tomkerkhove** [comment]
@christle Thanks a ton! Should we abandon the other PR then?

**Comment by christle** [comment]
If it is ok for you, then yes. It was easier for me to test & develop this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **Prometheus Scaler:** Introduce skipping of certificate check for unsigned certs ([#2310](https://github.com/kedacore/keda/issues/2310))
 - **Pulsar Scaler:** Add support for bearer token and basic auth ([#3844](https://github.com/kedacore/keda/issues/3844))
 - **Pulsar Scaler:** Add support for partitioned topics ([#3833](https://github.com/kedacore/keda/issues/3833))
+- **Eventhub Scaler:** Add new Checkpoint Strategy for Dapr ([#3022](https://github.com/kedacore/keda/issues/3022))
```
Comment (zroubalik) [inline CHANGELOG.md:74]:
```suggestion
- **Azure Eventhub Scaler:** Add new Checkpoint Strategy for Dapr ([#3022](https://github.com/kedacore/keda/issues/3022))
```

and please sort alphabetically

Comment (christle) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -65,6 +65,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **Azure Event Hub Scaler:** Support Azure Active Direcotry Pod & Workload Identity for Storage Blobs ([#3569](https://github.com/kedacore/keda/issues/3569))
 - **Azure Event Hub Scaler:** Support using connection strings for Event Hub namespace instead of the Event Hub itself. ([#3922](https://github.com/kedacore/keda/issues/3922))
 - **Azure Pipelines Scaler:** Improved speed of profiling large set of Job Requests from Azure Pipelines ([#3702](https://github.com/kedacore/keda/issues/3702))
+- **Azure Eventhub Scaler:** Add new Checkpoint Strategy for Dapr ([#3022](https://github.com/kedacore/keda/issues/3022))
```
Comment (tomkerkhove) [inline CHANGELOG.md:68]:
```suggestion
- **Azure Eventhub Scaler:** Support for `dapr` checkpoint strategy ([#3022](https://github.com/kedacore/keda/issues/3022))
- **Azure Pipelines Scaler:** Improved speed of profiling large set of Job Requests from Azure Pipelines ([#3702](https://github.com/kedacore/keda/issues/3702))
```

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (tomkerkhove) [reply CHANGELOG.md]: No worries; you are not alone in this!

**Pull request merged**